prefs.js

```diff
@@ -455,6 +458,50 @@ function createPrefsWidgetClass(resource_path, util) {
                     }
                 );
             }
+
+            copy_gnome_terminal_profile() {
+                // Check if gnome terminal is installed
+                const schema_source = Gio.SettingsSchemaSource.get_default();
+                if (schema_source.lookup('org.gnome.Terminal.ProfilesList', true) === null ||
+                    schema_source.lookup('org.gnome.Terminal.Legacy.Profile', true) === null) {
+                    const dialog = new Gtk.MessageDialog({
+                        transient_for: this.get_toplevel(),
+                        modal: true,
+                        buttons: Gtk.ButtonsType.CLOSE,
+                        message_type: Gtk.MessageType.ERROR,
+                        text: 'GNOME Terminal is not installed',
+                    });
+                    dialog.connect('response', () => dialog.destroy());
+                    dialog.show();
+
+                    return;
+                }
+
+                // Find default gnome terminal profile
+                let profiles_list = Gio.Settings.new('org.gnome.Terminal.ProfilesList');
+                let profilePath = profiles_list.settings_schema.get_path();
+                let uuid = profiles_list.get_string('default');
+                let gnome_terminal_profile = Gio.Settings.new_with_path(
+                    'org.gnome.Terminal.Legacy.Profile',
+                    `${profilePath}:${uuid}/`
+                );
+
+                // Copy color profile
+                [
+                    'use-theme-colors',
+                    'foreground-color', 'background-color',
+                    'bold-color-same-as-fg', 'bold-color',
+                    'cursor-colors-set', 'cursor-foreground-color', 'cursor-background-color',
+                    'highlight-colors-set', 'highlight-foreground-color', 'highlight-background-color',
+                    'palette',
+                    'bold-is-bright',
+                ].forEach(key => {
+                    this.settings.set_value(
+                        key,
+                        gnome_terminal_profile.get_value(key)
+                    );
```

Comment:
`.get_value()`: [It is a programmer error to give a key that isn't contained in the
schema](https://gjs-docs.gnome.org/gio20~2.66p/gio.settings#method-get_value)

`.set_value()`: [It is a programmer error to give a key that isn't contained in the
schema for this or for value to have the incorrect type](https://gjs-docs.gnome.org/gio20~2.66p/gio.settings#method-set_value)

"It is a programmer error" means the application may crash (like `Gio.Settings.new` with nonexistent schema), even if it currently doesn't. And GNOME Terminal's schemas may change.

So I think it's necessary to check that a) all keys exist in the profile schema b) values have expected (by ddterm settings schema) type.

Yep, I've missed multiple issues in the initial review